sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -0,0 +1,1501 @@
+        timeout: int = constants.DEFAULT_TIMEOUT,
+    ):
+        """List of all PyTorchJob in namespace.
+
+        Args:
+            namespace: Namespace to list the PyTorchJob.
+            timeout: Optional, Kubernetes API server timeout in seconds
+                to execute the request.
+
+        Returns:
+            list[KubeflowOrgV1PyTorchJob]: List of PyTorchJob objects. It returns
+            empty list if PyTorchJobs cannot be found.
+
+        Raises:
+            TimeoutError: Timeout to list PyTorchJobs.
+            RuntimeError: Failed to list PyTorchJobs.
+        """
+
+        return utils.list_jobs(
+            custom_api=self.custom_api,
+            api_client=self.api_client,
+            namespace=namespace,
+            job_model=models.KubeflowOrgV1PyTorchJob,
+            job_kind=constants.PYTORCHJOB_KIND,
+            job_plural=constants.PYTORCHJOB_PLURAL,
+            timeout=timeout,
+        )
+
+    def delete_pytorchjob(
+        self,
+        name: str,
+        namespace: str = utils.get_default_target_namespace(),
+        delete_options: client.V1DeleteOptions = None,
+    ):
+        """Delete the PyTorchJob
+
+        Args:
+            name: Name for the PyTorchJob.
+            namespace: Namespace for the PyTorchJob.
+            delete_options: Optional, V1DeleteOptions to set while deleting
+                the PyTorchJob. For example, grace period seconds.
+
+        Raises:
+            TimeoutError: Timeout to delete PyTorchJob.
+            RuntimeError: Failed to delete PyTorchJob.
+        """
+
+        utils.delete_job(
+            custom_api=self.custom_api,
+            name=name,
+            namespace=namespace,
+            job_kind=constants.PYTORCHJOB_KIND,
+            job_plural=constants.PYTORCHJOB_PLURAL,
+            delete_options=delete_options,
+        )
+
+    # ------------------------------------------------------------------------ #
+    # MXJob Training Client APIs.
+    # ------------------------------------------------------------------------ #
+    def create_mxjob(
+        self,
+        mxjob: models.KubeflowOrgV1MXJob,
+        namespace=utils.get_default_target_namespace(),
+    ):
+        """Create the MXJob.
+
+        Args:
+            mxjob: MXJob object of type KubeflowOrgV1MXJob.
+            namespace: Namespace for the MXJob.
+
+        Raises:
+            TimeoutError: Timeout to create MXJob.
+            RuntimeError: Failed to create MXJob.
+        """
+
+        utils.create_job(
+            custom_api=self.custom_api,
+            job=mxjob,
+            namespace=namespace,
+            job_kind=constants.MXJOB_KIND,
+            job_plural=constants.MXJOB_PLURAL,
+        )
+
+    def create_mxjob_from_func(self):
+        """Create MXJob from the function.
+        TODO (andreyvelich): Implement this function.
+        """
+        logging.warning("This API has not been implemented yet.")
```

Comment:
Would you like to work on this in this PR? Or another PR?

Reply:
I think, we can implement those in the following PRs.

Reply:
Sounds good.